dqc/api/properties.py

```diff
@@ -460,6 +483,28 @@ def _equadrupole(qc: BaseQCCalc) -> torch.Tensor:
 
     return quadrupole + ion_quadrupole
 
+@memoize_method
+def _optimal_geometry(qc: BaseQCCalc) -> torch.Tensor:
+    # calculate the optimal geometry
+    system = qc.get_system()
+    atompos = system.atompos
+
+    # check if the atompos requires grad
+    _check_differentiability(atompos, "atom positions", "hessian")
+
+    # get the energy for a given geometry
+    def _get_energy(atompos: torch.Tensor) -> torch.Tensor:
+        new_system = system.make_copy(moldesc=(system.atomzs, atompos))
+        new_qc = qc.__class__(new_system).run()
```

Comment:
Let me know if you don't like this - `qc.__class__` was the best I could think of

Reply:
That's fine. Thank you